crates/miden-proving-service-client/src/proving_service/batch_prover.rs

```diff
@@ -89,17 +94,39 @@ impl RemoteBatchProver {
 
         let request = tonic::Request::new(proposed_batch.into());
 
-        let response = client.prove(request).await.map_err(|err| {
-            RemoteProverError::other_with_source("failed to prove transaction", err)
-        })?;
+        let response = client
+            .prove(request)
+            .await
+            .map_err(|err| RemoteProverError::other_with_source("failed to prove block", err))?;
 
-        // Deserialize the response bytes back into a ProvenTransaction.
-        let proven_transaction = ProvenBatch::try_from(response.into_inner()).map_err(|_| {
-            RemoteProverError::other(
-                "failed to deserialize received response from remote transaction prover",
+        // Deserialize the response bytes back into a ProvenBatch.
+        let proven_batch = ProvenBatch::try_from(response.into_inner()).map_err(|err| {
+            RemoteProverError::other_with_source(
+                "failed to deserialize received response from remote prover",
+                err,
             )
         })?;
 
-        Ok(proven_transaction)
+        Ok(proven_batch)
+    }
+}
+
+// CONVERSIONS
+// ================================================================================================
+
+impl From<ProposedBatch> for ProvingRequest {
+    fn from(proposed_batch: ProposedBatch) -> Self {
+        ProvingRequest {
+            proof_type: ProofType::Batch.into(),
+            payload: proposed_batch.to_bytes(),
+        }
+    }
+}
+
+impl TryFrom<ProvingResponse> for ProvenBatch {
+    type Error = DeserializationError;
+
+    fn try_from(response: ProvingResponse) -> Result<Self, Self::Error> {
+        ProvenBatch::read_from_bytes(&response.payload)
     }
```

Comment:
Should we not check if the `proof_type` field here is set correctly?

Reply:
We do not have the proof_type in the response, it is just the payload.

Reply:
Ah! Then ignore this and other similar comments.